paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +662,98 @@ def validate(
             return []
 
 
-def load_vitess_instance_config(
+def generate_vitess_instance_config(
+    instance_config: Dict,
+) -> Dict:
+    # Generate the vitess instance config from the yelpsoa config
+
+    cpus = instance_config.get("cpus")
+    mem = instance_config.get("mem")
+    deploy_group = instance_config.get("deploy_group")
```

Comment:
a couple of these can probably use the LongRunningServiceConfigDict getters rather than directly reading the config dict (which would also have the benefit of already setting defaults for these)